.github/workflows/sync-reusable.yml

```diff
@@ -0,0 +1,135 @@
+# Inspired by https://github.com/SocketDev/workflows/blob/fe40aa6135a5ea6d81ad8103e4f044d2f9227e41/.github/workflows/reusable-sync.yml
+
+name: Reusable Sync
+
+on:
+  workflow_call:
+    secrets:
+      APP_PEM:
+        description: 'The private key of the GitHub App to use for the PR creation.'
+        required: false
+
+    inputs:
+      npm-sync-script:
+        description: 'The name of the script to run as the sync.'
+        required: true
+        type: string
+
+      branch-name:
+        description: 'The name of the branch to create'
+        required: true
+        type: string
+
+      pr-title:
+        description: 'The title of the PR.'
+        required: true
+        type: string
+      pr-body:
+        description: 'The body of the PR.'
+        required: true
+        type: string
+
+      commit-message:
+        description: 'The title of the PR.'
+        required: true
+        type: string
+
+# Push and PR permissions are handled using a bot app, see the "actions/create-github-app-token" step
+permissions:
+  contents: read
+
+jobs:
+  diff:
+    name: Sync ${{ inputs.npm-sync-script }}
+    environment: sync
+    runs-on: ubuntu-latest
+    steps:
+      # Set up repository
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          show-progress: false
+      - uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+
+      # New prefix generated every 120 days, to ensure npm cache doesn't grow too large
+      - name: Get npm cache timed expiration
+        id: npm-timed-expiration
+        run: echo "prefix=$(( $(date +%s) / 60 / 60 / 24 / 120 ))" >> $GITHUB_OUTPUT
+
+      - name: Determine npm cache directory
+        id: npm-cache-dir
+        run: echo "dir=$(npm config get cache)" >> $GITHUB_OUTPUT
+
+      - uses: actions/cache@v4
+        with:
+          path: ${{ steps.npm-cache-dir.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ steps.npm-timed-expiration.outputs.prefix }}-${{ hashFiles('main/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ steps.npm-timed-expiration.outputs.prefix }}-
+
+      - run: npm ci --ignore-scripts --force --no-fund --no-audit
+
+      # Actual syncing
+      - name: Prepare for diff
+        run: >
+          for PROJECT in $(jq -r 'join(" ")' dependents-data/projects.json); do
+            cp "dependents-data/$PROJECT.json" "dependents-data/$PROJECT-old.json"
+          done
+
+      - name: Run npm sync script
+        run: npm run ${{ inputs.npm-sync-script }}
```

Comment:
There are more types of syncs we will want to do eventually + a reusable workflow makes it much easier to separate intent from plumbing